fix(tui): Lint error in render-performance.test.tsx

## Description
PR #1020 introduced lint errors that break CI.

## Error
```
src/__tests__/benchmarks/render-performance.test.tsx
  58:3  error  Type number trivially inferred from a number literal, remove type annotation  @typescript-eslint/no-inferrable-types
```

## Fix
Line 58: Remove redundant type annotation from number literal.

Plus 8 warnings in useAdaptivePolling.ts (missing deps in useCallback).

## Priority
P1 - CI lint check failing

--- a/tui/src/__tests__/benchmarks/render-performance.test.tsx
+++ b/tui/src/__tests__/benchmarks/render-performance.test.tsx
@@ -55,7 +55,7 @@ function measureRenderTime(element: React.ReactElement): { time: number; output:
 // Utility to measure multiple renders and get statistics
 function benchmarkRender(
   element: React.ReactElement,
-  iterations: number = 10
+  iterations = 10
 ): { avg: number; min: number; max: number; p95: number } {
   const times: number[] = [];
 

